qa/abstracts/src/main/java/io/camunda/zeebe/process/test/qa/abstracts/assertions/AbstractProcessInstanceAssertTest.java

```diff
@@ -1435,4 +1439,47 @@ public void testShouldCaptureLatestValueOfVariable()
           .hasVariableWithValue(ProcessPackLoopingServiceTask.TOTAL_LOOPS, "3");
     }
   }
+
+  @Nested
+  class CustomMapperTests {
+
+    private ZeebeClient client;
+    private ZeebeTestEngine engine;
+
+    private final ObjectMapper objectMapper = configureObjectMapper();
+
+    @Test
+    public void shouldDeserializeDateVariables() throws InterruptedException, TimeoutException {
+      final Map<String, Object> variables = new HashMap<>();
+      variables.put("stringProperty", "stringValue");
+      variables.put("numberProperty", 123);
+      variables.put("booleanProperty", true);
+      variables.put("complexProperty", Arrays.asList("Element 1", "Element 2"));
+      variables.put("nullProperty", null);
+      variables.put("javaDate", LocalDateTime.of(2023, 8, 14, 16, 0, 0));
+      variables.put("stringDate", "\"2023-08-14T16:00:00\"");
```

Comment:
🔧 This would be a great candidate for a [parametrised test](https://www.baeldung.com/parameterized-tests-junit-5#6-method) 🙂

Reply:
It removes some complexity from the test. That way you don't need to create a Map in the test class. You don't need the foreach loop to go through all scenarios. And best of all, it will run all the cases even if one of them fails.

Reply:
Ok got it!